### ISSUE TITLE
Switch: interoperability with bootstrap.

### ISSUE THREAD

**Issue by tkrugg** [ibm-js/deliteful#493]
Switch was put in the sames situations as a bootstrap checkbox for comparison.
- [ ] various alignments issues (some are cause by labels wrapped inside div.checkbox, which might be a minor issue) 
- [ ] outline should follow validation state.
- [ ] When wrapped inside a label and is manually toggled by clicking on the label, it breaks. Strangely enough this doesn't happen _every_ time.

``` html
<label>
    <d-switch></d-switch>  Option 2
</label>
```

![image](https://cloud.githubusercontent.com/assets/2982512/5935741/6fd255ac-a6de-11e4-9d65-a24be90ba35c.png)
- [ ] when put inside an inline form, the widget looks broken
  ![image](https://cloud.githubusercontent.com/assets/2982512/5935763/b9cd10d4-a6de-11e4-92a5-3f16a90943ff.png)
- [ ] Also in inline forms, the height might need some tweaking to fit in nicely with other form elements.

[Link to manual test](http://tkrugg.github.io/bootstrap-deliteful/compiled/all-switch.html)

**Comment by buberdds** [comment]
Hey, what strategy you have to stength css selectors?

https://github.com/ibm-js/deliteful/blob/master/Switch/themes/Switch_template.less#L88
so you have (0010) and bootstrap is using input[type="checkbox"] (0011) and .checkbox input[type="checkbox"] (0021). If margin will be set 0, most of this cases will be fixed.

So, where this selector should be strengthened?

Switch_template.less
themes\bootstrap\Switch.less

thanks

**Comment by tkrugg** [comment]
nice catch @buberdds ! 
To me, it may make more sense to strengthen the CSS inside themes/bootstrap/switch.less if you don't expect users of the ios or holodark themes to load bootstrap.min.css. IMHO, they probably will so i'd say let's put it in switch_template.less.

**Comment by buberdds** [comment]
I looked also at this bug: 'when put inside an inline form, the widget looks broken'.

@media (min-width: 768px){  .form-inline .checkbox input[type=checkbox] ...  overrides position absolute. This is even stronger selector than pravious ones. Are you ok with selectors in css like: 

d-switch.d-switch  input[type=checkbox].-d-switch-input ? 
To be honest it looks terrible, so I started to think even about use of !important declarations ;)
I can make PR with anything, but maybe you have a different idea how we can fix this issues?

thx
